### DISCUSSION TITLE
Auto-migrating specific tables I define and ignoring all others?

### DISCUSSION THREAD

**Discussion by fazal-kh4n** [sqlalchemy/alembic#1007] [Usage Questions] [answered]
Hi,

I am working on a large project, which has several tables.

I want to ensure that all alembic actions, especially auto-migrations, are scoped to the tables I am working on.

To this end, I have edited my `env.py` file as follows (only key items highlighted):

```
my_schema_names = ['a', 'b', 'c']


def include_name(name, type_, parent_names):
    if type_ == "table":
        return name in my_schema_names
    else:
        return True

context.configure(
      url=url,
      target_metadata=target_metadata,
      literal_binds=True,
      dialect_opts={"paramstyle": "named"},
      include_schemas=True,
      include_name=include_name
)


target_metadata = Base.metadata
```

However, when I run the command:

`alembic revision --autogenerate -m "initial`

It produces version file that has operations on all the schemas that I do not want to deal with.

Have I misunderstood something? I just want to scope alembic to only deal with tables I care about and tell it to.

Thanks!

**Comment by zzzeek** [comment, marked as answer]
the "Type" being viewed should be "schema", not "table":

```py
def include_name(name, type_, parent_names):
    if type_ == "schema":
        return name in ["schema_one", "schema_two"]
    else:
        return True

context.configure(
    # ...
    include_schemas = True,
    include_name = include_name
)
```

https://alembic.sqlalchemy.org/en/latest/api/runtime.html?highlight=include_name#alembic.runtime.environment.EnvironmentContext.configure.params.include_name

**Reply by fazal-kh4n** [reply]
Thanks a lot @zzzeek - must have missed that!
